extension/js/common/browser/browser-msg.ts

```diff
@@ -250,6 +256,7 @@ export class BrowserMsg {
           BrowserMsg.sendAwait(undefined, 'getLocalKeyExpiration', bm, true) as Promise<Bm.Res.GetLocalKeyExpiration>,
       },
     },
+    confirmationResult: (dest: Bm.Dest, bm: Bm.ConfirmationResult) => BrowserMsg.sendCatch(dest, 'confirmation_result', bm),
```

Comment:
I added `confirmation_show` and `confirmation_result` messages to be able to relay confirmation display to the parent to make it "full-screen" and get the result back to the caller (similar to `passphrase_dialog` and `passphrase_entry`, only I used a promise instead of polling a structure with the result). It's not easy as the caller should have specific event listeners. For `AttachmentDownloadView` it works well with both webmail and `inbox.htm`.
What is your opinion, @sosnovsky ?

Reply:
Current implementation looks good, cleaner that polling, and it works well. Probably it's possible in the future to remake `passphrase_entry` to use promise too?

Reply:
Yeah, looks like not worth to change `confirmation` implementation for now, as current one works well